pkg/api/v1/kubernetes/clusters/clusters_test.go

```diff
@@ -186,67 +227,72 @@ func TestKubernetesClusterService_CreateOrUpdate(t *testing.T) {
 		args    args
 		wantRes clusters.Cluster
 		wantErr bool
+		useWait bool
 	}{
-		{"ctx is canceled", args{ctx_bad, projectID, clusterName, k_ok, np_ok, m_ok, h_ok, e_ok}, want, true},
-		{"bad project ID", args{ctx, "something", clusterName, k_ok, np_ok, m_ok, h_ok, e_ok}, want, true},
-		{"bad cluster name", args{ctx, projectID, "C_N-AME", k_ok, np_ok, m_ok, h_ok, e_ok}, want, true},
-		{"all ok", args{ctx, projectID, clusterName, k_ok, np_ok, m_ok, h_ok, e_ok}, want, false},
-		{"all ok 2", args{ctx, projectID, clusterName, k_ok, np_ok, nil, h_ok, e_ok}, want, false},
-		{"all ok 3", args{ctx, projectID, clusterName, k_ok, np_ok, nil, nil, e_ok}, want, false},
-		{"all ok 4", args{ctx, projectID, clusterName, k_ok, np_ok, nil, nil, nil}, want, false},
-
-		{"kube bad 0", args{ctx, projectID, clusterName, k_bad_0, np_ok, m_ok, h_ok, e_ok}, want, true},
-		{"kube bad 1", args{ctx, projectID, clusterName, k_bad_1, np_ok, m_ok, h_ok, e_ok}, want, true},
-
-		{"np bad 0", args{ctx, projectID, clusterName, k_ok, np_bad_0, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 1", args{ctx, projectID, clusterName, k_ok, np_bad_1, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 2", args{ctx, projectID, clusterName, k_ok, np_bad_2, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 3", args{ctx, projectID, clusterName, k_ok, np_bad_3, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 4", args{ctx, projectID, clusterName, k_ok, np_bad_4, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 5", args{ctx, projectID, clusterName, k_ok, np_bad_5, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 6", args{ctx, projectID, clusterName, k_ok, np_bad_6, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 7", args{ctx, projectID, clusterName, k_ok, np_bad_7, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 8", args{ctx, projectID, clusterName, k_ok, np_bad_8, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 9", args{ctx, projectID, clusterName, k_ok, np_bad_9, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 10", args{ctx, projectID, clusterName, k_ok, np_bad_10, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 11", args{ctx, projectID, clusterName, k_ok, np_bad_11, m_ok, h_ok, e_ok}, want, true},
-		{"np bad 12", args{ctx, projectID, clusterName, k_ok, np_bad_12, m_ok, h_ok, e_ok}, want, true},
-
-		{"maintenance bad 1", args{ctx, projectID, clusterName, k_ok, np_ok, m_bad_1, h_ok, e_ok}, want, true},
-		{"maintenance bad 2", args{ctx, projectID, clusterName, k_ok, np_ok, m_bad_2, h_ok, e_ok}, want, true},
-		{"maintenance bad 3", args{ctx, projectID, clusterName, k_ok, np_ok, m_bad_3, h_ok, e_ok}, want, true},
-
-		{"h bad 1", args{ctx, projectID, clusterName, k_ok, np_ok, m_ok, h_bad_1, e_ok}, want, true},
-		{"h bad 2", args{ctx, projectID, clusterName, k_ok, np_ok, m_ok, h_bad_2, e_ok}, want, true},
-
-		{"e bad 1", args{ctx, projectID, clusterName, k_ok, np_ok, m_ok, h_ok, e_bad_1}, want, true},
+		{"ctx is canceled", args{ctx_bad, projectID, clusterName, k_ok, np_ok, m_ok, h_ok, e_ok}, want, true, false},
```

Comment:
Given that the tests all use the same `want`, you could consider keeping it out of the test arguments and just using `want` directly in the testing logic. For tt.projectID you could also consider using the zero value when it has the default, so something like `args{Context: ctx, ClusterName: clusterName, ...}` where it's the default, and `args{Context: ctx, ProjectID: "something", ClusterName: clusterName, ...}` where it's not. Then you could in the test write `if tt.ProjectID == "" {tt.ProjectID = projectID}`, but that's a matter of taste.

Reply:
want is only created if there's no error (otherwise it's nil)
so using tt.useWait makes sense